### ISSUE TITLE
S3's `check-path` is not specific enough

### ISSUE THREAD

**Issue by kennytm** [pingcap/br#898]
Please answer these questions before submitting your issue. Thanks!

### 1. What did you do?

1. dump some SQL files into `s3://some-bucket/some-prefix/` on AWS S3.
2. create a role, give it the `s3:ListBucket` and `s3:GetObject` permission, conditioned with `s3:prefix`=`some-prefix/`.
3. run Lightning using that role

### 2. What did you expect to see?

successfully restored.

### 3. What did you see instead?

failed due to an unnecessary `HeadBucket` call:

```
tidb lightning encountered error:  create storage failed: Bucket some-bucket is not accessible: Forbidden: Forbidden
        status code: 403, request id: ..., host id: ...: [BR:ExternalStorage:ErrStorageInvalidConfig]invalid external storage config
```

### 4. What version of BR and TiDB/TiKV/PD are you using?

master

**Comment by kennytm** [comment]
Note: affects all users of `br/pkg/storage`, including BR itself (can workaround with `--skip-check-path`), TiCDC and Dumpling (both have SkipCheckPath hard-coded to false).

**Comment by handlerww** [comment]
Reproduce:
1 Create IAM policy, select S3 permission with GetObject and ListBucket, add a request condition about s3:prefix.

![image](https://user-images.githubusercontent.com/2058782/111681583-5dfb8480-885e-11eb-8293-e5db3f9f4774.png)

2. Create a IAM role, attach the policy created before to this role

3. Set IAM role of EC2 to a certain role
    
![image](https://user-images.githubusercontent.com/2058782/111680492-4079eb00-885d-11eb-92ff-3cca0c5c083c.png)

**Comment by lonng** [comment]
@glorv Seems this bug had been fixed in #899, can we close this issue?

**Comment by glorv** [comment]
closed in favor of https://github.com/pingcap/dumpling/pull/286 and https://github.com/pingcap/ticdc/pull/1890
